### PR TITLE
[VarDumper] Fix tests for PHP 8.1

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Part of #41552 
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Component/VarDumper/Tests/Caster/ReflectionCasterTest.php [hunk 1]
```diff
@@ -122,17 +122,31 @@ public function testReflectionParameter()
     {
         $var = new \ReflectionParameter(reflectionParameterFixture::class, 0);
 
-        $this->assertDumpMatchesFormat(
-            <<<'EOTXT'
+        if (80100 > \PHP_VERSION_ID) {
```
Comment (Nyholm) [inline src/Symfony/Component/VarDumper/Tests/Caster/ReflectionCasterTest.php]: This is just nitpicking to be consistent with the rest of the code. 

Please write `\PHP_VERSION_ID < 80100`

Same thing below.

**Comment by derrabus** [comment]
Can you explain why the dumper behaves differently on PHP 8.1?

**Comment by stof** [comment]
@derrabus looks like the behavior of `ReflectionGenerator::getTrace()` changed in PHP 8.1, returning some info when there was none before.

**Comment by alexandre-daubois** [comment]
@derrabus also, some other functions seem to have changed their behaviour. For example, when `null`, `ReflectionParameter::getDefaultValue()` returns null on PHP 8.0, and `ReflectionException: Internal error: Failed to retrieve the default value` on PHP 8.1. Searched why in recent PRs of https://github.com/php/php-src/, but couldn't find which commit is responsible of it.

File: src/Symfony/Component/VarDumper/Tests/Caster/ReflectionCasterTest.php (hunk 1 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/VarDumper/Tests/Caster/ReflectionCasterTest.php]: Done! But aren't we using yoda-style conditions in this repository?

Comment (stof) [reply src/Symfony/Component/VarDumper/Tests/Caster/ReflectionCasterTest.php]: there is no variable involved here

Comment (alexandre-daubois) [reply src/Symfony/Component/VarDumper/Tests/Caster/ReflectionCasterTest.php]: Understood, thanks for the explanation!

**Comment by derrabus** [comment]
> @derrabus For example, when `null`, `ReflectionParameter::getDefaultValue()` returns null on PHP 8.0, and `ReflectionException: Internal error: Failed to retrieve the default value` on PHP 8.1.

I'd like to clarify first, if that change was intentional: https://bugs.php.net/bug.php?id=81147

**Comment by derrabus** [comment]
The other changes look reasonable to me.

**Comment by derrabus** [comment]
Okay, that was fast and we've got the answer we were looking for.

> Just to confirm, this is indeed an intentional change to ensure that optional-before-required is consistently treated as a required parameter. Previously it was treated as such in nearly all situations, but there were loopholes. One, as you have discovered, is reflection. The other is named arguments, where it was possible to not specify such a parameter if later required parameters were specified.

Can you have a look at the Travis failure? It looks related.

**Comment by alexandre-daubois** [comment]
Thank you so much for your investigation, super good to know! I'll definitely take a look at Travis tomorrow to finish this PR. 👍

**Comment by alexandre-daubois** [comment]
@derrabus done :ok_hand: [This failing build](https://travis-ci.com/github/symfony/symfony/jobs/515175456) for 7.4 seems unrelated to this PR.

I'm not sure I understand the difference between "Tests / Integration (x.x)" and Travis CI tests ?

**Review by derrabus — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by derrabus** [comment]
Thank you @alexandre-daubois.

**Pull request merged**